freud/data.py

```diff
@@ -212,10 +209,14 @@ def make_random_system(box_size, num_points, is2D=False, seed=None):
             Generated box and points.
     """  # noqa: E501
     if seed is not None:
+        random_state = np.random.get_state()
```

Comment:
Does there need to be a `try/finally` block to ensure that the state is reset in the event of failure mid-function?